### PR TITLE
Add segment and span context implementation

### PR DESCRIPTION
**Pull request by Shikugawa**
It includes

- [x] SpanContext definition
- [x] SegmentContext definition
- [x] Span which belongs to current SegmentContxt definition
- [x] Native context definition -> data collect protocol protobuf definition for SegmentObject and SpanObject
- [ ] Add `sw8-x` capability
- [x] Add unit tests

### CODE REVIEW

File: source/segment_context.cc [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Rei Shimizu
```
Comment (wu-sheng) [inline source/segment_context.cc:1]: Just as a reminder, if this is set as copyright by individual, please make sure the following CI would check this.

I don't want this project ends by copyright from various people.

Also, it is common, copyright of the SkyAPM virtual org. It is up to you.

Comment (Shikugawa) [reply source/segment_context.cc]: Changed from my name to `SkyAPM`. right?

Comment (wu-sheng) [reply source/segment_context.cc]: If that makes sense to you, my point is, we need a consistent header for all contributors in the future, and maybe we want to donate it to the ASF later.

Comment (Shikugawa) [reply source/segment_context.cc]: Sure. Thank you for pointing out. For now, we should suit them as `SkyAPM`. When we'd like to donate this to ASF, we will change to `Apache Software Foundation (ASF)`.

Comment (wu-sheng) [reply source/segment_context.cc]: Yes. That is the thing after the donation.

**Comment by wu-sheng** [comment]
You added these tests for local coding only only right? Note the GitHub CI has not been set

**Comment by Shikugawa** [comment]
@wu-sheng yep. I will introduce GitHub based CI. (only for execute test)

**Review by dio — COMMENTED** [review]
Looks good modulo test, I prefer we have an integration repo with Envoy and have integration tests there.

**Pull request merged**